### ISSUE TITLE
http/1.0 is not being handled correctly.

### ISSUE THREAD

**Issue by ijunaid8989** [keathley/finch#111]
I am doing this in mint.

```
{:ok, conn} = Mint.HTTP.connect(:http, "93.107.118.128", 8083)
{:ok, conn, request_ref} = Mint.HTTP.request(conn, "GET", "/snapshot.jpg", [{"Authorization", "Basic junaid="}], "")
```

as a result of first, we get this

```elixir
{:ok,
 %Mint.HTTP1{
   buffer: "",
   host: "93.107.118.128",
   mode: :active,
   port: 8083,
   private: %{},
   request: nil,
   requests: {[], []},
   scheme_as_string: "http",
   socket: #Port<0.7>,
   state: :open,
   transport: Mint.Core.Transport.TCP
 }}
```
and on 2nd we get this.
```elixir
{:ok,
 %Mint.HTTP1{
   buffer: "",
   host: "93.107.118.128",
   mode: :active,
   port: 8083,
   private: %{},
   request: %{
     body: nil,
     connection: [],
     content_length: nil,
     data_buffer: [],
     headers_buffer: [],
     method: "GET",
     ref: #Reference<0.2499171616.2477260803.176902>,
     state: :status,
     status: nil,
     transfer_encoding: [],
     version: nil
   },
   requests: {[], []},
   scheme_as_string: "http",
   socket: #Port<0.7>,
   state: :open,
   transport: Mint.Core.Transport.TCP
 }, #Reference<0.2499171616.2477260803.176902>}
```

when I do this.

```elixir
{:ok, conn, responses} = receive do msg -> Mint.HTTP.stream(conn, msg) end
```
I get this 
```elixir
{:ok,
 %Mint.HTTP1{
   buffer: "",
   host: "93.107.118.128",
   mode: :active,
   port: 8083,
   private: %{},
   request: %{
     body: :until_closed,
     connection: [],
     content_length: nil,
     data_buffer: [[] | ""],
     headers_buffer: [],
     method: "GET",
     ref: #Reference<0.2499171616.2477260803.176902>,
     state: :body,
     status: 200,
     transfer_encoding: [],
     version: {1, 0}
   },
   requests: {[], []},
   scheme_as_string: "http",
   socket: #Port<0.7>,
   state: :open,
   transport: Mint.Core.Transport.TCP
 },
 [
   {:status, #Reference<0.2499171616.2477260803.176902>, 200},
   {:headers, #Reference<0.2499171616.2477260803.176902>,
    [
      {"server", "HTTPD"},
      {"pragma", "no-cache"},
      {"cache-control", "no-cache"},
      {"content-type", "image/jpeg"}
    ]}
 ]}
```
on-again doing this
```elixir
 {:ok, conn, responses} = receive do msg -> Mint.HTTP.stream(conn, msg) end
```

I get this as response

```elixir
{:ok,
 %Mint.HTTP1{
   buffer: "",
   host: "93.107.118.128",
   mode: :active,
   port: 8083,
   private: %{},
   request: %{
     body: :until_closed,
     connection: [],
     content_length: nil,
     data_buffer: [],
     headers_buffer: [],
     method: "GET",
     ref: #Reference<0.2499171616.2477260803.176902>,
     state: :body,
     status: 200,
     transfer_encoding: [],
     version: {1, 0}
   },
   requests: {[], []},
   scheme_as_string: "http",
   socket: #Port<0.7>,
   state: :open,
   transport: Mint.Core.Transport.TCP
 },
 [
   {:data, #Reference<0.2499171616.2477260803.176902>,
    <<255, 216, 255, 224, 0, 16, 74, 70, 73, 70, 0, 1, 1, 0, 0, 1, 0, 1, 0, 0,
      255, 219, 0, 67, 0, 13, 9, 10, 11, 10, 8, 13, 11, 10, 11, 14, 14, 13, 15,
      19, 32, 21, 19, ...>>}
 ]}
```

but when I do the same request in Finch

```elixir
Finch.build(:get, "http://93.107.118.128:8083/snapshot.jpg", [{"Authorization", "Basic junaid="}]) |> Finch.request(MyFinch)
```

I just get this.

```elixir
{:error, %Mint.TransportError{reason: :closed}}
```

**Comment by ericmj** [comment]
Turns out this is a Mint issue, I will open a PR to Mint with a fix.

**Comment by whatyouhide** [comment]
I released Mint v1.2.1 with the fix in https://github.com/elixir-mint/mint/pull/290 :)
